category/category.go

```diff
@@ -0,0 +1,157 @@
+package category
+
+import (
+	"context"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/almighty/almighty-core/log"
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
+)
+
+// Defines "type" string to be used while validating jsonapi spec based payload
+const (
+	APIStringTypeCategory = "categories"
+)
+
+const (
+	PlannerRequirements = "planner.requirements"
+	PlannerIssues       = "planner.issues"
+)
+
+var (
+	PlannerRequirementsID = uuid.FromStringOrNil("04aef834-1505-44cf-80e4-ab0d857d9f56") // "planner.requirements"
+	PlannerIssuesID       = uuid.FromStringOrNil("27d92fe4-b2ee-45c2-b9bb-01f355ad616f") // "planner.issues"
+)
+
+// Category describes a single category
+type Category struct {
+	gormsupport.Lifecycle
+	ID   uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	Name string
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m *Category) TableName() string {
+	return "categories"
+}
+
+// Category_wit_relationship describes relationship between a category and a workitemtype.
+type CategoryWitRelationship struct {
```

Comment:
I was worried that the object name would be too long. Anyway I will change it to `CategoryWorkItemTypeRelationship`.